src/System.Text.Json/src/System/Text/Json/Serialization/MemberAccessor.cs

```diff
@@ -0,0 +1,93 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Reflection;
+using System.Runtime.CompilerServices;
+
+namespace System.Text.Json.Serialization
+{
+    internal static class MemberAccessor
+    {
+        private delegate TProperty GetProperty<TClass, TProperty>(TClass obj);
+        private delegate TProperty GetPropertyByRef<TClass, TProperty>(ref TClass obj);
+
+        private delegate void SetProperty<TClass, TProperty>(TClass obj, TProperty value);
+        private delegate void SetPropertyByRef<TClass, TProperty>(ref TClass obj, TProperty value);
+
+        private delegate Func<object, TProperty> GetPropertyByRefFactory<TClass, TProperty>(GetPropertyByRef<TClass, TProperty> set);
+        private delegate Action<object, TProperty> SetPropertyByRefFactory<TClass, TProperty>(SetPropertyByRef<TClass, TProperty> set);
+
+        private static readonly MethodInfo s_createStructPropertyGetterMethod = new GetPropertyByRefFactory<int, int>(CreateStructPropertyGetter)
+            .Method.GetGenericMethodDefinition();
+
+        private static readonly MethodInfo s_createStructPropertySetterMethod = new SetPropertyByRefFactory<int, int>(CreateStructPropertySetter)
+            .Method.GetGenericMethodDefinition();
+
+        internal static Func<object, TProperty> CreatePropertyGetter<TClass, TProperty>(PropertyInfo propertyInfo)
+        {
+            MethodInfo getMethodInfo = propertyInfo.GetGetMethod();
+
+            if (typeof(TClass).IsValueType)
+            {
+                var factory = CreateDelegate<GetPropertyByRefFactory<TClass, TProperty>>(s_createStructPropertyGetterMethod.MakeGenericMethod(typeof(TClass), typeof(TProperty)));
```

Comment:
nit: Don't use var here and other places that you can't statically infer the type from the right-hand-side of the expression.

Reply:
It can be statically inferred from the right side. The type is `GetPropertyByRefFactory<TClass, TProperty>`.

Reply:
Really? Doesn't the return type depend on the implementation of `CreateDelegate<T>` which just happens to return the same type as the generic. It could very well be returning a string and hence requires knowledge about the implementation to infer the type (i.e. the reviewer has to go find the implementation :p).

However, it's just a nit. Feel free to ignore :)